### ISSUE TITLE
Keep alive on websocket

### ISSUE THREAD

**Issue by kevinvalk** [tweedegolf/mailcrab#53]
If you tab away from mailcrab for a few minutes (4+ minutes) it will not receive any new emails automatically and it will require a refresh. Even if you stay in the tab for a hot minute it will not show new mails. The websocket seems to be in limbo.

Tested with Chrome and MailCrab v0.16

**Comment by stappersg** [comment]
On Tue, Apr 18, 2023 at 01:13:13PM -0700, Kevin Valk wrote:
> ... it will require a refresh ...
> 
> Tested with Chrome and MailCrab v0.16

Please tell more about that setup.


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by kevinvalk** [comment]
I put it to a different prefix http://localhost:8001/smtp/, but as far as I can see there is no ping/pong keep alive mechanism on the websocket. So it is a matter of time (phun intended) before something (webbrowser, server, etc) times out and closes the websocket. Normally this is solved with a periodic ping pong on the socket to prevent any timeouts. Maybe because I host mailcrab on a different prefix ping/pong is not working?


```
caught (in promise) Error: A listener indicated an asynchronous response by returning true, but the message channel closed before a response was received
```

**Comment by marlonbaeten** [comment]
I haven't experienced a websocket timeout myself, but maybe I'm sending many messages ;-)

Sure - It seems useful to implement a heartbeat mechanism. I'll add that 👍🏼

**Comment by stappersg** [comment]
For what it is worth: I do hope that `mailcrab` remains a _catch all email test server_, that _SPA_, Single Page Application, is just a "nice to have".

**Comment by marlonbaeten** [comment]
Release [v0.17](https://github.com/tweedegolf/mailcrab/releases/tag/v0.17.0) sends out a ping message over the websocket connection every 30 seconds.

**Comment by stappersg** [comment]
![image](https://user-images.githubusercontent.com/35434094/233863614-45e7417d-d020-4d0f-a7ac-287057f2e027.png)

**Comment by stappersg** [comment]
On Sun, Apr 23, 2023 at 01:17:29PM -0700, Geert Stappers wrote:
> ![image](https://user-images.githubusercontent.com/35434094/233863614-45e7417d-d020-4d0f-a7ac-287057f2e027.png)

Text version:
  It can be seen that issue reporter
  has seen the "it is fixed" report

**Comment by stappersg** [comment]
I had also seen _"it is fixed"_ and tried it, but didn't see that it is fixed.

Now I wonder if original issue reporter has **seen** it fixed.

**Comment by stappersg** [comment]
Meanwhile I learnt about https://github.com/tweedegolf/mailcrab#path-prefix
